src/components/MMSearch/MMSearch.component.tsx

```diff
@@ -0,0 +1,211 @@
+
+        return () => {
+            document.body.removeEventListener('click', onDropDownCloseHandler);
+        };
+    }, []);
+
+    /**
+     * If 'isOpen' is true and 'value' is empty, then set the active index to 0 and scroll the list to (0,0)
+     */
+    useEffect(() => {
+        if (open) {
+            setActive(0);
+            if (typeof listRef.current?.scrollTo === 'function') {
+                listRef.current.scrollTo(0, 0);
+            }
+        }
+    }, [open]);
+
+    /**
+     * The function is called when an event is detected on the keyboard,
+     * so you can browse through the list and select one.
+     */
+    const onKeyDown: React.KeyboardEventHandler<HTMLInputElement> = (event) => {
+        if (!filteredOptions.length || !open) {
+            return;
+        }
+        if (event.key === 'Enter') {
+            event.preventDefault();
+            const option = filteredOptions[active];
+
+            if (onSelect) {
+                onSelect(event, option);
+            }
+
+            setActive(0);
+            return;
+        }
+        if (event.key === 'ArrowUp') {
+            event.preventDefault();
+            if (active === 0) {
+                return;
+            }
+            setActive((prev) => prev - 1);
+            if (typeof listRef.current?.scrollBy === 'function') {
+                listRef.current.scrollBy(0, -Constants.ITEM_HEIGHT);
+            }
+            if (inputRef.current) {
+                inputRef.current.focus();
+            }
+            return;
+        }
+        if (event.key === 'ArrowDown') {
+            event.preventDefault();
+            if (active === filteredOptions.length - 1) {
+                return;
+            }
+            setActive((prev) => prev + 1);
+            if (typeof listRef.current.scrollBy === 'function') {
+                listRef.current.scrollBy(0, Constants.ITEM_HEIGHT);
+            }
+        }
+    };
+
+    return (
+        <AutoCompleteWrapper
+            fullWidth={fullWidth}
+            className={`mm-autocomplete ${className}`}
+        >
+            <Input
+                ref={inputRef}
+                fullWidth={fullWidth}
+                searchQuery={searchQuery}
+                onKeyDown={onKeyDown}
+                value={searchValue}
+                label={label}
+                iconName={leadingIcon}
+                onClose={() => {
+                    if (onClearInput) {
+                        onClearInput();
+                    }
+                    if (inputRef?.current) {
+                        inputRef.current.focus();
+                    }
+                    setOpen(true);
+                    setSearchValue('');
+                    setSearchQuery('');
+                }}
+                onChange={(e) => {
+                    setOpen(true);
+                    setSearchQuery(e.target.value);
+                    setSearchValue(e.target.value);
+                }}
```

Comment:
Don't use inline function if the function definition is more than 2 lines.

Reply:
Fixed